packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -393,6 +393,9 @@ export default defineRenderHandler(async (event): Promise<Partial<RenderResponse
     for (const style in styles) {
       const resource = styles[style]
       if (!import.meta.dev || !isRenderingIsland || (resource.file.includes('scoped') && !resource.file.includes('pages/'))) {
+        // support for the ?inline query parameter for Vite version > 5.0
+        if (process.env.NUXT_VITE_NODE_OPTIONS && resource.file.endsWith("?inline")) break;
```

Comment:
```suggestion
        if (import.meta.dev && resource.file.endsWith('?inline')) continue
```

Reply:
we should probably also parse the url in case there are other queries using `getQuery` from `ufo`...

Reply:
yeah, we really should parse the url. But, I don't know much about `h3` and `ufo`, you can switch to their api.